### PR TITLE
ROC-3826: Add testing support endpoint for linking defendant to case

### PR DESCRIPTION
**Pull request by filipczyks**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java [hunk 1]
```diff
@@ -163,10 +163,34 @@ public void linkDefendant(String authorisation) {
...
+    }
+
     private void linkToCase(User defendantUser, User anonymousCaseWorker, String letterHolderId, String caseId) {
         String defendantId = defendantUser.getUserDetails().getId();
+
         LOGGER.info("Granting access to case: {} for user: {} with letter-holder id: {}",
             caseId, defendantId, letterHolderId);
+        this.grantAccessToCase(anonymousCaseWorker, caseId, defendantId);
+
+        LOGGER.info("Revoking access to case: {} for user: {}", caseId, letterHolderId);
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: I'd add info about letter holder to make it easier to understand. I t took mi a bit ti figure out what is going on.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java [hunk 2]
```diff
@@ -57,9 +64,24 @@ public Claim updateRespondByDate(
...
+    @PutMapping("/claims/{claimReferenceNumber}/defendant/{defendantUsername}")
+    public Claim linkDefendantToClaim(
+        @PathVariable("claimReferenceNumber") String claimReferenceNumber,
+        @PathVariable("defendantUsername") String defendantUsername,
+        @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authorisation
+    ) {
+        Claim claim = getClaim(claimReferenceNumber, authorisation);
+
+        //todo how we should store default password? const? properties?
+        User defendant = this.userService.authenticateUser(defendantUsername, "Password12");
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: @dharmendrak would it be possible to update the case as anonymous caseworker?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java [hunk 3]
```diff
@@ -57,9 +64,24 @@ public Claim updateRespondByDate(
...
+
+        supportRepository.linkDefendantToClaim(claim, defendant);
+
+        return getClaim(claimReferenceNumber, authorisation);
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: We do not have to return claim back. Can we please remove it as we want to stop doing it in CCD world. If there is a need for frontend to retrieve claim there will be explicit call.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java (hunk 2 above)
Comment (geek-soft) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: @dharmendrak ☝️ 

The story was created on the assumption that anonymous caseworker will perform the operation and there won't be user JWT required.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java [hunk 4]
```diff
@@ -57,9 +64,23 @@ public Claim updateRespondByDate(
...
 
+    @PutMapping("/claims/{claimReferenceNumber}/defendant/{defendantId}")
+    public Claim linkDefendantToClaim(
+        @PathVariable("claimReferenceNumber") String claimReferenceNumber,
+        @PathVariable("defendantId") String defendantId,
+        @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authorisation
+    ) {
+        Claim claim = getClaim(claimReferenceNumber, authorisation);
+
+        //todo how we should store default password? const? properties?
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: please resolve TODO

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java [hunk 5]
```diff
@@ -163,10 +163,32 @@ public void linkDefendant(String authorisation) {
             ).forEach(caseId -> linkToCase(defendantUser, anonymousCaseWorker, letterHolderId, caseId)));
     }
 
+    public void linkDefendantToCaseByCaseId(String caseId, String defendantId) {
+        User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+        this.linkToCaseWithoutRevoking(defendantId, anonymousCaseWorker, caseId);
+    }
+
+    private void linkToCaseWithoutRevoking(String defendantId, User anonymousCaseWorker, String caseId) {
+        LOGGER.info("Granting access to case: {} for user: {}", caseId, defendantId);
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: Do all these events need to be at info?
debug level I think would be more appropriate

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java (hunk 4 above)
Comment (filipczyks) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: resolved, we no longer authenticate users, so there's no need to have password here

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java (hunk 5 above)
Comment (geek-soft) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: Done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java (hunk 4 above)
Comment (geek-soft) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/IntegrationTestSupportController.java]: I removed TODO comment as well

**Pull request merged**